jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -417,21 +420,45 @@ public JibContainerBuilder setWorkingDirectory(@Nullable AbsoluteUnixPath workin
   public JibContainer containerize(Containerizer containerizer)
       throws InterruptedException, ExecutionException, IOException,
           CacheDirectoryCreationException {
-    BuildConfiguration buildConfiguration = toBuildConfiguration(containerizer);
-    BuildResult result = containerizer.getTargetImage().toBuildSteps(buildConfiguration).run();
-    return new JibContainer(result.getImageDigest(), result.getImageId());
+
+    return containerize(containerizer, Executors::newCachedThreadPool);
+  }
+
+  @VisibleForTesting
+  JibContainer containerize(
+      Containerizer containerizer, Supplier<ExecutorService> executorServiceFactory)
```

Comment:
Would consider a different name for this `executorServiceFactory` since it's just a backup/default.

Reply:
maybe the method needs to be renamed to indicate the `Supplier` is an or else kind of situation?